From 86de902e35a7c57ecc008ba725b7842e36bf7393 Mon Sep 17 00:00:00 2001
From: Dave Young <dyoung@redhat.com>
Date: Wed, 15 Feb 2012 14:40:24 +0800
Subject: [PATCH] ssh-client module install fix

commit 38164332e198f7cc8f339b42d555796918de04b3 try to
disallow ssh-client install for without proper options.
But ssh-client will fail installation with --sshkey for
publickey mode

Fix it by refusing to install only when both --ctty and --sshkey
are not added in options.

Signed-off-by: Dave Young <dyoung@redhat.com>
---
 modules.d/95ssh-client/module-setup.sh |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/95ssh-client/module-setup.sh b/modules.d/95ssh-client/module-setup.sh
index 4410e4e..8d62b2d 100755
--- a/modules.d/95ssh-client/module-setup.sh
+++ b/modules.d/95ssh-client/module-setup.sh
@@ -50,8 +50,8 @@ inst_sshenv()
 }
 
 install() {
-    [[ ! $cttyhack = yes ]] && {
-        derror "ssh interactive mode needs option --ctty!"
+    [[ ! $cttyhack = yes && ! $sshkey ]] && {
+        derror "ssh-client needs option --ctty or --sshkey!"
         return 1
     }
 
